kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/validators/SchemaValidator.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.homeoffice.digital.sas.kafka.validators;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_VALIDATED;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_COMMA_DELIMETER;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_JSON_ATTRIBUTE;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Getter
+@Setter
+@NoArgsConstructor
+@Component
+public class SchemaValidator {
+
+  @Value("${kafka.supported.schema.version}")
```

Comment:
now that this is a spring bean, would be better to autowire this through constructor injection instead of field injection for flexibility.
E.g. move this annotation next to the param in a new constructor.